packages/typespec-autorest-canonical/src/emitter.ts

```diff
@@ -84,7 +84,7 @@ async function emitAllServices(
     validateUnsupportedVersioning(program, service.type);
     const context: AutorestEmitterContext = {
       program,
-      outputFile: resolveOutputFile(service, services.length > 1, options),
+      outputFile: resolveOutputFile(service, services.length > 1, options, canonicalVersion),
```

Comment:
wouldnt it be better to hard code this, instead of treating this as a "version"

Reply:
I removed this version parameter and passed the constant "canonical" to interpolatePath(). Can I ask what do you mean "hard code in the config"? Thanks

Reply:
i mean why are we interpolating "cannonical" as a version, why is it allowed in the output file, the default value should just have canonical included instead of `{version}`

Reply:
I agree with that. I removed that parameter for resolveOutputFile and just used "canonical" in it.

Reply:
I mean there too it shouldn't the user should NOT be allowd to pass `{version}` in output file and have that be interpolated. change the default value of `output-file`

Reply:
update the default of output-file